com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/dao/APIDAO.java

```diff
@@ -882,4 +882,52 @@ public static boolean inAPIQuotaDateRange(String serviceProvider,String applicat
 		return inQuotaDateRange;
 	}
 
+	public boolean isBlacklistedorWhitelistedNumber(String msisdn, String apiId,
+													String appId, String subscriberId,
+													String action) throws Exception {
+
+		Connection connection = null;
+		PreparedStatement preparedStatement = null;
+		ResultSet resultSet = null;
+		boolean isListed = false;
+
+		try {
+			StringBuilder queryBuilder = new StringBuilder();
+
+			queryBuilder.append("select * FROM ");
```

Comment:
Will be clearer if you could execute the query to check the data existence straightaway at the SQL level without selecting data and checking the existence in java level. Check the com.wso2telco.dep.operatorservice.dao.BlackListWhiteListDAO#searchBlacklistIdByMsisdn method

Reply:
Changed the SQL query to get a boolean value